flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -337,8 +337,7 @@ private static TaskSlotTable createTaskSlotTable(
 	private static Map<MemoryType, Long> calculateMemorySizeByType(
 			TaskManagerServicesConfiguration taskManagerServicesConfiguration) {
 		final Map<MemoryType, Long> memorySizeByType = new HashMap<>();
-		memorySizeByType.put(MemoryType.HEAP, taskManagerServicesConfiguration.getOnHeapManagedMemorySize().getBytes());
-		memorySizeByType.put(MemoryType.OFF_HEAP, taskManagerServicesConfiguration.getOffHeapManagedMemorySize().getBytes());
+		memorySizeByType.put(MemoryType.OFF_HEAP, taskManagerServicesConfiguration.getManagedMemorySize().getBytes());
```

Comment:
I think we should completely remove
`calculateMemorySizeByType`, `Map<MemoryType, Long> memorySizeByType` and the second `computeSlotResourceProfile`
in `TaskManagerServices`.